### PR TITLE
TINY-11273: add zero width nobreak space to isWhitespace function

### PR DESCRIPTION
**Pull request by ztomaszyk**
Related Ticket: TINY-11273

Description of Changes:
When formatting is being applied a span with `zero width no-break space (ufeff)` is added to the editor. This character was not considered a whitespace and because of that the autocompleter was not triggering properly. To fix this bug I added the character to the `isWhitespace` function. 

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~~Docs ticket created (if applicable)~~

- [x] Don't merge until we make the 7.4 RC

GitHub issues (if applicable):

### CODE REVIEW

**Review by TheSpyder — APPROVED** [review]

**Review by lorenzo-pomili — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/autocomplete/AutocompleteTest.ts [hunk 1]
```diff
@@ -766,4 +766,12 @@ describe('browser.tinymce.themes.silver.editor.autocomplete.AutocompleteTest', (
     },
     assertion: (editor) => TinyAssertions.assertContent(editor, '<p>multi-bB</p>')
   }));
+
+  it('TINY-11273: Autocomplete should be triggered for content with formatting applied.', async () => {
```
Comment (spocke) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/autocomplete/AutocompleteTest.ts:770]: I think we generally don't have a dot at the end of the test labels.

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/autocomplete/AutocompleteTest.ts [hunk 2]
```diff
@@ -766,4 +766,12 @@ describe('browser.tinymce.themes.silver.editor.autocomplete.AutocompleteTest', (
     },
     assertion: (editor) => TinyAssertions.assertContent(editor, '<p>multi-bB</p>')
   }));
+
+  it('TINY-11273: Autocomplete should be triggered for content with formatting applied', async () => {
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/test/ts/browser/editor/autocomplete/AutocompleteTest.ts:770]: Can we add another formatting test like `bold` to make sure the fix works for spans and other inline tags like `strong`

File: .changes/unreleased/tinymce-TINY-11273-2024-09-24.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Fixed
+body: Autocompleter would not activate after font size change.
```
Comment (ltrouton) [inline .changes/unreleased/tinymce-TINY-11273-2024-09-24.yaml:3]: I am wondering if this should be more generic
```suggestion
body: Autocompleter would not activate after applying an inline format like font size in some cases.
```

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/themes/silver/test/ts/browser/editor/autocomplete/AutocompleteTest.ts (hunk 2 above)
Comment (ztomaszyk) [reply modules/tinymce/src/themes/silver/test/ts/browser/editor/autocomplete/AutocompleteTest.ts]: Added ✅

**Pull request merged**